test/controllers/rentals_controller_test.rb

```diff
@@ -0,0 +1,155 @@
+      expect(body).must_be_instance_of Array
+      body.each do |rental|
+        expect(rental).must_be_instance_of Hash
+        expect(rental.keys.sort).must_equal RENTAL_FIELDS
+      end
+    end
+
+    it "returns an empty array if not rentals exist" do
+      Rental.destroy_all
+      # Act
+      get rentals_path
+
+      # getting the body. JSON.parse: command take that json turn into a ruby array or hash.
+      body = JSON.parse(response.body)
+      #Assert
+      expect(body).must_be_instance_of Array
+      expect(body.length).must_equal 0
+    end
+  end
+
+  describe "check out" do 
+    let(:rental_data) {
+      {
+        video_id: videos(:video_1).id,
+        customer_id: customers(:customer_1).id
+      }
+    }
+
+    it "can allow check out" do
+      inventory_count = videos(:video_1).available_inventory
+      video_count = customers(:customer_1).videos_checked_out_count
+
+      expect {
+        post check_out_path, params: rental_data
+      }.must_differ "Rental.count", 1
+
+      must_respond_with :success
+
+      video = Video.find_by(id: rental_data[:video_id])
+      customer = Customer.find_by(id: rental_data[:customer_id])
+
+      expect(video.available_inventory).must_equal inventory_count - 1
+      expect(customer.videos_checked_out_count).must_equal video_count + 1
+    end
+
+    it "prevents checkout if customer does not exist" do
+      rental_data[:customer_id] = nil
+
+      expect {
+        post check_out_path, params: rental_data
+      }.must_differ "Rental.count", 0
+
+      must_respond_with :not_found
+    end
+
+    it "prevents checkout if video does not exist" do
+      rental_data[:video_id] = nil
+
+      expect {
+        post check_out_path, params: rental_data
+      }.must_differ "Rental.count", 0
+
+      must_respond_with :not_found
+    end
+
+    it "prevents checkout if video is not available" do
+      rental_data[:video_id] = videos(:video_2).id
+
+      expect {
+        post check_out_path, params: rental_data
+      }.must_differ "Rental.count", 0
+
+      must_respond_with :bad_request
+    end
+  end
+
+  describe "check in" do 
+    let(:rental_data) {
+      {
+        video_id: videos(:video_1).id,
+        customer_id: customers(:customer_1).id
+      }
+    }
+
+    before do 
+      post check_out_path, params: rental_data
+    end
+
+    it "can allow check in" do
+      inventory_count = videos(:video_1).available_inventory
+      video_count = customers(:customer_1).videos_checked_out_count
+
+      expect {
+        post check_in_path, params: rental_data
+      }.must_differ "Rental.count", 0
+
+      must_respond_with :success
+      expect(Rental.all.first.check_in_date).must_equal Date.today
+
+      video = Video.find_by(id: rental_data[:video_id])
+      customer = Customer.find_by(id: rental_data[:customer_id])
+
+      expect(video.available_inventory).must_equal inventory_count
+      expect(customer.videos_checked_out_count).must_equal video_count
```

Comment:
I think I might be missing something but I'm confused why these numbers aren't meant to change when the user checks in a rental.

Reply:
From looking at the source code in the controller, I'm surprised and confused why this passes. I must be reading something wrong.